subsys/net/openthread/rpc/server/ot_rpc_dns_client.c

```diff
@@ -0,0 +1,763 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_common.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+
+#include <zephyr/net/openthread.h>
+
+static void ot_rpc_dns_client_get_default_config(const struct nrf_rpc_group *group,
+						 struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	const otDnsQueryConfig *config;
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+
+	config = otDnsClientGetDefaultConfig(openthread_get_default_instance());
+
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	NRF_RPC_CBOR_ALLOC(group, rsp_ctx, OT_RPC_DNS_QUERY_CONFIG_SIZE);
+
+	ot_rpc_encode_dns_query_config(&rsp_ctx, config);
```

Comment:
Since `config` is a pointer to OT-managed object, it's probably better to call this before unlocking the mutex.

Reply:
Good point, fixed